jasypt/src/main/java/org/jasypt/encryption/pbe/StandardPBEByteEncryptor.java

```diff
@@ -811,7 +830,7 @@ public synchronized void initialize() {
                 } catch (final Exception e) {
                     // If encryption fails, it is more secure not to return any 
                     // information about the cause in nested exceptions. Simply fail.
-                    throw new EncryptionOperationNotPossibleException();
+                    throw new EncryptionOperationNotPossibleException(e);
```

Comment:
I think we either run their code or run our code.

Reply:
The interfaces (between theirs and ours) are not compatible because of `setAlgorithm(String secretKeyFactoryAlgorithm, String cipherAlgorithm)`. Switching between the versions requires code change or a new adapter class to invoke this method via reflection.